activesupport/lib/active_support/number_helper/number_to_currency_converter.rb

```diff
@@ -43,6 +41,18 @@ def i18n_opts
           i18n[:negative_format] ||= "-#{i18n[:format]}" if i18n[:format]
           i18n
         end
+
+        def fails_to_parse?
+          # Float() is stricter than String.to_f, but is slightly slower because it raises an
+          # exception in those cases. We should use this method with care to avoid impacting
+          # performance.
+          Float(self.number)
+          false
+        rescue ArgumentError
+          # likely an alternate input format that failed to parse.
+          # see https://github.com/rails/rails/pull/39350
+          true
+        end
```

Comment:
TIL you can also suppress exceptions for Kernel#Float:
https://rubyapi.org/3.0/o/kernel#method-i-Float

```ruby
Float("123.0_badstring", exception: false)  #=> nil
```

```suggestion
        def fails_to_parse?
          !Float(self.number, exception: false)
        end
```